cpu/samd5x/periph/can.c

```diff
@@ -0,0 +1,768 @@
+            return -1;
+    }
+
+    return 0;
+}
+
+static void _setup_clock(can_t *dev)
+{
+    if (dev->conf->can == CAN0) {
+        GCLK->PCHCTRL[CAN0_GCLK_ID].reg = GCLK_PCHCTRL_CHEN | GCLK_PCHCTRL_GEN(dev->conf->gclk_src);
+    }
+    else if (dev->conf->can == CAN1) {
+        GCLK->PCHCTRL[CAN1_GCLK_ID].reg = GCLK_PCHCTRL_CHEN | GCLK_PCHCTRL_GEN(dev->conf->gclk_src);
+    }
+    else {
+        DEBUG_PUTS("CAN channel not supported");
+    }
+}
+
+static void _set_bit_timing(can_t *dev)
+{
+    assert(dev->candev.bittiming.sjw >= 1);
+    assert(dev->candev.bittiming.phase_seg2 >= 1);
+    assert(dev->candev.bittiming.phase_seg1 + dev->candev.bittiming.prop_seg >= 1);
+    assert(dev->candev.bittiming.brp >= 1);
+
+    DEBUG("bitrate=%" PRIu32 ", sample_point=%" PRIu32 ", brp=%" PRIu32 ", prop_seg=%" PRIu32
+          ", phase_seg1=%" PRIu32 ", phase_seg2=%" PRIu32 ", sjw=%" PRIu32 "\n", dev->candev.bittiming.bitrate, dev->candev.bittiming.sample_point,
+          dev->candev.bittiming.brp, dev->candev.bittiming.prop_seg, dev->candev.bittiming.phase_seg1, dev->candev.bittiming.phase_seg2, dev->candev.bittiming.sjw);
+
+    /* Set bit timing */
+    dev->conf->can->NBTP.reg = (uint32_t)((CAN_NBTP_NTSEG2(dev->candev.bittiming.phase_seg2 - 1))
+                            | (CAN_NBTP_NTSEG1(dev->candev.bittiming.phase_seg1 + dev->candev.bittiming.prop_seg - 1))
+                            | (CAN_NBTP_NBRP(dev->candev.bittiming.brp - 1))
+                            | (CAN_NBTP_NSJW(dev->candev.bittiming.sjw - 1)));
+}
+
+static void _set_tx_fifo_data_size(can_t *dev, uint8_t size) {
+    assert(size < 0x8);
+    dev->conf->can->TXESC.reg |= CAN_TXESC_TBDS(size);
+}
+
+static void _set_rx_buffer_data_size(can_t *dev, uint8_t size) {
+    assert(size < 0x8);
+    dev->conf->can->RXESC.reg |= CAN_RXESC_RBDS(size);
+}
+
+static void _set_rx_fifo_0_data_size(can_t *dev, uint8_t size) {
+    assert(size < 0x8);
+    dev->conf->can->RXESC.reg |= CAN_RXESC_F0DS(size);
+}
+
+static void _set_rx_fifo_1_data_size(can_t *dev, uint8_t size) {
+    assert(size < 0x8);
+    dev->conf->can->RXESC.reg |= CAN_RXESC_F1DS(size);
+}
+
+static void _set_can_pins(can_t *dev)
+{
+    assert(dev->conf->tx_pin != GPIO_UNDEF);
+    assert(dev->conf->rx_pin != GPIO_UNDEF);
+
+    gpio_init(dev->conf->tx_pin, GPIO_OUT);
+    gpio_init(dev->conf->rx_pin, GPIO_IN);
+
+    if (dev->conf->can == CAN0) {
+        gpio_init_mux(dev->conf->tx_pin, GPIO_MUX_I);
+        gpio_init_mux(dev->conf->rx_pin, GPIO_MUX_I);
+    }
+    else if (dev->conf->can == CAN1) {
+        gpio_init_mux(dev->conf->tx_pin, GPIO_MUX_H);
+        gpio_init_mux(dev->conf->rx_pin, GPIO_MUX_H);
+    }
+    else {
+        DEBUG_PUTS("Unsupported can channel");
+    }
+}
+
+void candev_samd5x_enter_sleep_mode(candev_t *candev)
+{
+    can_t *dev = container_of(candev, can_t, candev);
+
+    dev->conf->can->CCCR.reg |= CAN_CCCR_CSR;
+    while(!(dev->conf->can->CCCR.reg & CAN_CCCR_CSA));
```

Comment:
please check the  "GitHub Actions / static-tests"  results 

full block {} expected in the control structure

keyword 'while' not followed by a single space

Reply:
I'm sorry I don't get it. The static tests are successful as far as I see from the CI checks.

Reply:
the riot static test produce errors and warnings, so you got no errors but warnings about the coding style 

someone decided  that riot needs empty blocks -> there is a rule to test that 
also there needs to be a space after while 

you can see these warnings when you open the changed files tab 

there are also some lines that are longer than 100 chars, if you can break them you should do so 

these warnings are very annoying since they will show up often if the file is changed in future

Reply:
the run will show the warnings as well https://github.com/RIOT-OS/RIOT/actions/runs/8065564951